schema.graphql

```diff
@@ -115,51 +162,98 @@ type DecimalValue @entity {
+  notionalVolFeesSweptUSD: BigDecimal!
+
+  # Clearinghouse pointer
+  clearinghouse: Clearinghouse!
+  # Metrics for each Token on same day
+  tokenDayDatas: [TokenDayData!]! @derivedFrom(field: "clearinghouseDayData")
 }
 
 type Token @entity {
+  # Token Address
   id: ID!
+
+  # Token Info
   symbol: String!
   name: String!
-  decimals: BigInt!
-  totalValueLocked: BigDecimal!
-  totalValueLockedUSD: BigDecimal!
+  decimals: Int!
+
+  # Total number of tokens in Clearinghouse
+  totalValueLocked: BigInt!
+
+  # Amount of tokens that are ready to be swept as fees
+  feeBalance: BigInt!
+  # Lifetime sum of tokens paid to Valorem as fees
+  feesAccrued: BigInt!
+
+  # Daily metrics (tvl, volume, etc) for the token
+  tokenDayDatas: [TokenDayData!]! @derivedFrom(field: "token")
 }
 
 type TokenDayData @entity {
-  # token address concatendated with date
+  # Token Address + Timestamp of day at 00:00AM UTC
   id: ID!
-  # timestamp rounded to current day by dividing by 86400
+  # Timestamp of day at 00:00AM UTC
   date: Int!
-  # pointer to token
+
+  # The Token that these metrics belong to
   token: Token!
-  # tvl across all contracts in token units
-  totalValueLocked: BigDecimal!
-  # tvl across all contracts in usd
-  totalValueLockedUSD: BigDecimal!
-  # token volume in token units
-  volume: BigDecimal!
-  # token volume in usd
-  volumeUSD: BigDecimal!
+
+  # TVL in Clearinghouse in token units
+  totalValueLocked: BigInt!
+
+  ## Notional Volumes in token units
+  notionalVolWritten: BigInt! # Underlying Asset written
+  notionalVolExercised: BigInt! # Exercise Asset exercised for Options
+  notionalVolRedeemed: BigInt! # Underlying and/or Exercise Asset redeemed from Claims
+  notionalVolTransferred: BigInt! # Claim Transferred: Current Underlying and/or Exercise Asset; Option Transferred: Underlying Asset
+  notionalVolSettled: BigInt! # Fee Earning Volume; Written + Exercised
+  notionalVolSum: BigInt! # Written + Exercised + Redeemed + Transferred
+  notionalVolFeesAccrued: BigInt!
+  notionalVolFeesSwept: BigInt!
+
+  ## Notional Volumes in USD
+  notionalVolWrittenUSD: BigDecimal!
+  notionalVolExercisedUSD: BigDecimal!
+  notionalVolRedeemedUSD: BigDecimal!
+  notionalVolTransferredUSD: BigDecimal!
+  notionalVolSettledUSD: BigDecimal!
+  notionalVolSumUSD: BigDecimal!
+  notionalVolFeesAccruedUSD: BigDecimal!
+  notionalVolFeesSweptUSD: BigDecimal!
+
+  # Pointer to ClearinghouseDayData for same day
+  clearinghouseDayData: ClearinghouseDayData!
 }
 
-type FeeSwitch @entity {
-  # OptionSettlementEngine contract address
+type Clearinghouse @entity {
+  # Clearinghouse contract address
   id: ID!
-  # recipient of fees
-  feeToAddress: String!
-  isEnabled: Boolean!
+
+  # Whether or not fees for protocol are enabled
+  feesEnabled: Boolean!
+  # The address of the recipient of protocol fees
+  feeToAddress: Account!
+
+  # Daily metrics
+  clearinghouseDayDatas: [ClearinghouseDayData!]!
```

Comment:
data is already plural. HistoricalDayData works

Reply:
:+1: 
fixed here https://github.com/valorem-labs-inc/valorem-subgraph/pull/43/commits/f1be2e1a5d3eac6e222d336be498649b79daacea